### DISCUSSION TITLE
megalinter flavors grouping

### DISCUSSION THREAD

**Discussion by vquie** [renovatebot/renovate#23861] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.42.4

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I've got megalinter-ci_light with semver docker versioning in a gitlab-ci.yaml file and megalinter (full) in a makefile for manual usage. The first is using built in gitlab ci handling, the latter uses matchString.
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "regexManagers": [
    {
      "fileMatch": [
        "Makefile"
      ],
      "matchStrings": [
        "#\\s*renovate:\\s*datasource=(?<datasource>[^ ]*)\\s*depName=(?<depName>[^ ]*)(?:\\s*versioning=(?<versioning>[^ ]*))?\\s+_ML_VERSION\\s=\\s(?<currentValue>[^\n\r]*)"
      ],
      "versioningTemplate": "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}"
    }
  ]
}
```

This is the snippet in my MAKEFILE:
```
# Mega-Linter Version
# renovate: datasource=docker depName=oxsecurity/megalinter versioning=docker
_ML_VERSION = v6
```

I want to group multiple megalinter flavors from different files to a single PR even if one is patch and the other is major only. Currently it it creates a different PR for each.

### What you tried so far.

```
{
  "packageRules": [
    {
      "matchSourceUrls": "https://github.com/oxsecurity/megalinter",
      "groupName": "MegaLinter",
      "groupSlug": "megalinter"
    }
  ]
}
```

I also tried this:
```
{
  "packageRules": [
    {
      "matchPackagePatterns": [
        "^oxsecurity/megalinter.*"
      ],
      "groupName": "MegaLinter",
      "groupSlug": "megalinter"
    }
  ]
}
```

**Comment by rarkins** [comment]
This config is wrong:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "packageRules": [
    {
      "fileMatch": [
        "Makefile"
      ],
      "matchStrings": [
        "#\\s*renovate:\\s*datasource=(?<datasource>[^ ]*)\\s*depName=(?<depName>[^ ]*)(?:\\s*versioning=(?<versioning>[^ ]*))?\\s+_ML_VERSION\\s=\\s(?<currentValue>[^\n\r]*)"
      ],
      "versioningTemplate": "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}"
    }
  ]
}
```

The options fileMatch, matchStrings and versioningTemplate belong in a regex manager, not packageRules

**Reply by vquie** [reply]
I made a mistake when copying the data here. This would be the correct snippet:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "regexManagers": [
    {
      "fileMatch": [
        "Makefile"
      ],
      "matchStrings": [
        "#\\s*renovate:\\s*datasource=(?<datasource>[^ ]*)\\s*depName=(?<depName>[^ ]*)(?:\\s*versioning=(?<versioning>[^ ]*))?\\s+_ML_VERSION\\s=\\s(?<currentValue>[^\n\r]*)"
      ],
      "versioningTemplate": "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}"
    }
  ]
}
```

This is working as there are two PR created.
1. chore(deps): update oxsecurity/megalinter-ci_light docker tag to v7.3.0 # via gitlabci in the .gitlab-ci.yaml
2. chore(deps): update oxsecurity/megalinter docker tag to v7 # via regex in the Makefile

**Reply by rarkins** [reply]
Please update your original question/post and make sure it's clear what you're asking. I don't understand why you pasted this snippet (even if incorrectly) and then had something _different_ for "what I tried so far".

**Reply by vquie** [reply]
I updated the original post.
In "Wanted end result." I posted the config that is working and explained what I wanted to achieve. Since I am mixing gitlabci with regex managers I thought it was important to solve this.
In "What you tried so far." I posted the config snippets I tried to achieve the result which are not working, thus "tried so far".

**Reply by rarkins** [reply]
So to group together updates you need them to have the same branch name. By default, you'll get the same branch name if the `depName` is the same and so too is the major release. If you're getting separate PRs then your choice is either to change the depName in your regex manager (note: you can have different depName and packageName values), or to group them manually as you tried.

If you remove the grouping rule you attempted, what are the two branch names you get by default?

**Reply by vquie** [reply]
Thanks.

The branches I get with my `matchSourceUrls` grouping attempt are
`renovate/major-megalinter`
`renovate/megalinter`

Without grouping the created branches are
`renovate/oxsecurity-megalinter-ci_light-7.x`
`renovate/oxsecurity-megalinter-7.x`

**Reply by rarkins** [reply]
When grouping, those two branches are expected and correct - one for major and one for non-major. Is there any problem with that?

**Reply by vquie** [reply, marked as answer]
Your hint with the `depName` helped me a lot. I removed the grouping and updated the `depName` in the Makefile. It is suggesting the wrong image but the correct versions. I can work with it as it is now.

![ml](https://github.com/renovatebot/renovate/assets/21147643/53366996-7269-48d3-9bf9-a561a92c0e0d)

Thanks.
